opsdroid/events.py

```diff
@@ -0,0 +1,169 @@
+"""Classes to describe different kinds of possible event."""
+
+from abc import ABC
+from datetime import datetime
+from copy import copy
+import asyncio
+from random import randrange
+
+from opsdroid.helper import get_opsdroid
+
+
+# pylint: disable=too-few-public-methods
+class Event(ABC):
+    """A generic event type.
+
+    Initiates an Event object with the most basic information about its
+    creation.
+
+    Args:
+        user: String name of user sending event
+        room: String name of the room or chat channel in which event was sent
+        connector: Connector object used to interact with given chat service
+        raw_event: Optional raw representation for event.
+
+    Attributes:
+        created: Local date and time that message object was created
+        user: String name of user sending message
+        room: String name of the room or chat channel in which message was sent
+        connector: Connector object used to interact with given chat service
+        raw_event: Raw event provided by chat service
+        responded_to: Boolean initialized as False. True if event has been
+            responded to
+
+    """
+
+    def __init__(self, user, room, connector, raw_event=None):  # noqa: D107
+        self.created = datetime.now()
+        self.user = user
+        self.room = room
+        self.connector = connector
+        self.raw_event = raw_event
+        self.responded_to = False
+
+
+class Message(Event):
+    """A message object.
+
+    Stores messages in a format that allows OpsDroid to respond or react with
+    delays for thinking and typing as defined in configuration YAML file.
+
+    Args:
+        user: String name of user sending message
+        room: String name of the room or chat channel in which message was sent
+        connector: Connector object used to interact with given chat service
+        text: String text of message
+        raw_event: Raw message as provided by chat service. None by default
+
+    Attributes:
+        created: Local date and time that message object was created
+        user: String name of user sending message
+        room: String name of the room or chat channel in which message was sent
+        connector: Connector object used to interact with given chat service
+        text: Text of message as string
+        raw_event: Raw message provided by chat service
+        raw_match: A match object for a search against which the message was
+            matched. E.g. a regular expression or natural language intent
+        responded_to: Boolean initialized as False. True if event has been
+            responded to
+
+    """
+
+    def __init__(self, user, room, connector,
+                 text, raw_event=None):  # noqa: D107
+        """Create object with minimum properties."""
+        super().__init__(user, room, connector)
+        self.text = text
+        self.raw_event = raw_event
```

Comment:
Could we also reinstate `raw_message` here to avoid this being a breaking change. It can just point to `self.raw_event`.